Sources/Segment/Plugins/SegmentDestination.swift

```diff
@@ -113,47 +114,50 @@ public class SegmentDestination: DestinationPlugin, Subscriber {
     }
     
     public func flush() {
```

Comment:
It seems like this could be simplified to:
```
public func flush() {
    flushQueue.async { [weak self] in
        self?._flush()
    }
}

// Existing flush (no need for this comment-- just making the point)
private func _flush() {
  // existing code...
}
```